doc/subsystems/networking/net-app-api.rst

```diff
@@ -0,0 +1,199 @@
+.. _net_app_api:
+
+Network Application API
+#######################
+
+The Network Application API allows applications to
+
+* **Initialize** the application for networking use. This means for example
+  that if the application needs to have an IPv4 address, and if DHCPv4 is
+  enabled, then the net-app API will make sure that the device will get an
+  IPv4 address before the application is started.
+
+* **Set** various options for the networking subsystem. This means that if the
+  user has set options like IP addresses, IEEE 802.15.4 channel etc. in the
+  project configuration file, then those settings are applied to the system
+  before application starts.
+
+* **Create** a simple TCP/UDP server or client application. The net-app API
+  has functions that make it easy to create a simple TCP or UDP based network
+  application. The net-app API also provides transparent TLS and DTLS support
+  for the application.
+
+The net-app API functionality is enabled by `CONFIG_NET_APP` option.
```

Comment:
Yes, use

     ``CONFIG_NET_APP``

if your intention is to have it show up as an inline code block with fixed-width font and the grey background.  If you want this to be a reference to the documentation for the CONFIG_NET_APP Kconfig option, you can use

     :option:`CONFIG_NET_APP`